### PR TITLE
Templates

### PR DESCRIPTION
**Pull request by mrbuds**
# Description

- Use the new Cooldown Swipe & Edge options for spells with charges on Icon region

preview

- fix "Lightning Call" buff template

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Tested radiance & lava burst for charges
- [x] Tested Lightning Call with the 2 talents required to make it relevant (storm elemental + primal elementalist)

# Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

**Comment by Stanzilla** [comment]
Is that 1:1 how it looks like on Blizzard buttons?

**Comment by mrbuds** [comment]
blizzard: preview

**Comment by InfusOnWoW** [comment]
Yes this matches how Blizzard action bars look like, but:
* The description claims that the icon is greyed out at 0 charges, that's no longer the case
* The fix for Call Lightning is wrong. This will show the Call Lightning from any player.

We need to either extend the ownOnly to consider buffs/debuffs from the pet as "ownOnly". I think that might be the easiest solution and probably not too bad.

Or extend the "ownOnly" from a boolean to a "ownOnly/ownOnlyAndPet/ownPet/anyone selection. That doesn't feel necessary.

**Comment by mrbuds** [comment]
For simplicity sake i removed the "turn grey on zero charge" from description, since it's region dependent and i don't think that part of the description is a key for user choice.

File: WeakAuras/BuffTrigger2.lua [hunk 1]
```diff
@@ -1884,18 +1884,34 @@ local function createScanFunc(trigger)
...
-        return false
-      end
-    ]]
-  elseif trigger.ownOnly == false then
-    ret = ret .. [[
-      if matchData.unitCaster == 'player' then
-        return false
-      end
-    ]]
+  if trigger.unit == "pet" then
```
Comment (InfusOnWoW) [inline WeakAuras/BuffTrigger2.lua]: that's wrong. .unit is the target. OwnOnly is about the source.

File: WeakAurasOptions/BuffTrigger2.lua [hunk 2]
```diff
@@ -404,9 +404,15 @@ local function GetBuffTriggerOptions(data, optionTriggerChoices)
       end,
       desc = function()
         local value = trigger.ownOnly
```
Comment (InfusOnWoW) [inline WeakAurasOptions/BuffTrigger2.lua]: Same here, don't change depending on trigger.unit. That's the target.

File: WeakAuras/BuffTrigger2.lua [hunk 3]
```diff
@@ -1886,13 +1886,13 @@ local function createScanFunc(trigger)
 
   if trigger.ownOnly then
     ret = ret .. [[
-      if matchData.unitCaster ~= 'player' then
+      if matchData.unitCaster ~= 'player' and matchData.unitCaster ~= 'pet' then
```
Comment (InfusOnWoW) [inline WeakAuras/BuffTrigger2.lua]: OR

**Pull request merged**